File: union_find/java/TestDisjointSet.java

public class TestDisjointSet
{
	public static void testUnionFind()
	{
		DisjointSet ds = new DisjointSet();
		for(int i = 0; i < 100; i++)
		{
			ds.makeSet(i);
			ds.union(i, i % 5);
		}
		for(int i = 0; i < 100; i++)
			for(int j = 0; j < 100; j++)
				if((i % 5 == j % 5) != (ds.findSet(i) == ds.findSet(j)))
					System.out.println("Error in Union-Find");
	}
	public static void main(String[] args)
	{
		testUnionFind();
	}
}

File: disjoint_set/py3/TestDisjointSet.py
import unittest;
from DisjointSet import DisjointSet
from DisjointSet import DisjointSetLinkBySize

class TestDisjointSet(unittest.TestCase):
    def testUnionFind(self):
        ds = DisjointSet(range(100))
        for i in range(100):
            ds.union(i, i % 5)
        for i in range(100):
            for j in range(i, 100):
                self.assertEqual(ds.find_set(i) == ds.find_set(j), i%5 ==j%5)

    def testDisjointSetLinkBySize(self):
        ds = DisjointSetLinkBySize(range(100))
        for i in range(100):
            ds.union(i, i % 5)
        for i in range(100):
            for j in range(i, 100):
                self.assertEqual(ds.find_set(i) == ds.find_set(j), i%5 ==j%5)

if __name__ == '__main__':
    unittest.main()
